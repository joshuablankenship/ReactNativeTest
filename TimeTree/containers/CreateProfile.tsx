import React, {Component} from 'react';
import {Switch} from 'react-native';
import {Image, Text, TextInput, TouchableOpacity, View} from 'react-native';
import {styles} from '../styles/CreateProfileStyles';
import MyDatePicker from '../components/DatePicker';
interface Props {
  navigation: any;
  openProfileSheet: any;
  closeProfileSheet: any;
  closeSignupSheet: any;
  closeLoginSheet: any;
}
interface State {
  name: string;
  password: string;
  isPopupTrue: boolean;
  displayBirthday: boolean;
  birthday: string;
}

export default class CreateProfile extends Component<Props, State> {
  constructor(props: Props | Readonly<Props>) {
    super(props);
    this.state = {
      name: '',
      password: '',
      isPopupTrue: false,
      displayBirthday: false,
      birthday: 'Not Set',
    };
  }
  _onPressButtonGoogle = async () => {
    this.props.navigation.navigate('Main');
  };
  _onPressButton = async () => {
    this.props.navigation.navigate('Main');
  };
  _openProfileSheet = () => {
    this.props.openProfileSheet();
  };
  _closeProfileSheet = () => {
    this.props.closeProfileSheet();
  };
  _toggle = () => {
    this.setState(prevState => ({
      displayBirthday: !prevState.displayBirthday,
    }));
  };
  setBirthday = (text: string) => {
    this.setState({
      birthday: text,
    });
  };

  render() {
    return (
      <View style={styles.sheetHeader}>
        <Text style={styles.title}> Edit Profile</Text>
        <Image
          style={styles.image}
          source={require('../assets/images/profile.png')}
        />
        <View style={styles.buttonContainer}>
          <View style={styles.textinputLight}>
            <Text style={styles.text}>Name</Text>
          </View>
          <View style={styles.modalButtons}>
            <TextInput
              style={styles.textinput}
              placeholder="Untitled"
              value={this.state.name}
              autoCorrect={false}
              onChangeText={text => this.setState({name: text})}
            />
            {this.state.name.length > 0 ? (
              <TouchableOpacity onPress={() => this.setState({name: ''})}>
                <Image
                  style={styles.a_logo}
                  source={require('../assets/images/close-button-png.png')}
                />
              </TouchableOpacity>
            ) : (
              <View />
            )}
          </View>
          <View style={styles.textinputLight}>
            <Text style={styles.text}>Birthday</Text>
          </View>
          <MyDatePicker
            birthday={this.state.birthday}
            setBirthday={this.setBirthday}
          />
          <View style={styles.textinput}>
            <Text style={styles.text}>Display Birthday</Text>
            <Switch
              style={styles.switchStyle}
              trackColor={{true: '#2FCC87', false: 'grey'}}
              thumbColor={'#f8f8f8'}
              ios_backgroundColor={
                this.state.displayBirthday ? '#2FCC87' : 'gray'
              }
              onValueChange={this._toggle}
              value={this.state.displayBirthday}
            />
          </View>
        </View>
        <View style={styles.bottom}>
          <TouchableOpacity
            disabled={
              this.state.name.length === 0 || this.state.birthday === 'Not Set'
            }
            style={styles.buttonGreen}
            onPress={() => {
              this.setState({isPopupTrue: true});
              setTimeout(() => {
                this._closeProfileSheet();
                this.props.closeSignupSheet();
                this.props.closeLoginSheet();
              }, 1000);
            }}>
            <Text
              style={
                this.state.name.length === 0 ||
                this.state.birthday === 'Not Set'
                  ? styles.textDark
                  : styles.buttonText
              }>
              Confirm
            </Text>
          </TouchableOpacity>
        </View>
        {this.state.isPopupTrue && (
          <View style={styles.overlay}>
            <Image
              style={styles.imageSmall}
              source={require('../assets/images/loading.gif')}
            />
          </View>
        )}
      </View>
    );
  }
}
