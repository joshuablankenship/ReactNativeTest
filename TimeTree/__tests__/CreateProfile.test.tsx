import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import {Image, Switch, TouchableOpacity} from 'react-native';
import CreateProfile from '../containers/CreateProfile';
const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  closeLoginSheet: jest.fn(),
  ...props,
});
describe('Test Create Profile Screen', () => {
  let wrapper: ShallowWrapper;
  let props: any;
  wrapper = shallow(<CreateProfile {...props} />);
  props = createTestProps({
    navigate: () => {},
    closeLoginSheet: () => {},
  });
  it('It should match initial snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('It should display three touchable opacity components', () => {
    expect(wrapper.find(TouchableOpacity).length).toEqual(1);
  });
  it('It should display one profile image', () => {
    expect(wrapper.find(Image).length).toEqual(1);
  });
  it('It should contain one Switch', () => {
    expect(wrapper.find(Switch).length).toEqual(1);
  });
});
