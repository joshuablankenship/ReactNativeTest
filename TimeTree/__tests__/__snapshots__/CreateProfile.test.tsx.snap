// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Create Profile Screen It should match initial snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "color": "#2FCC87",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 20,
        "fontWeight": "bold",
        "justifyContent": "center",
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     Edit Profile
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/profile.png",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "center",
        "marginBottom": 50,
        "marginTop": 30,
        "width": 200,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 10,
        "width": "75%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "borderColor": "#f8f8f8",
          "borderWidth": 1,
          "fontWeight": "bold",
          "height": 53,
          "padding": 10,
          "paddingTop": 15,
          "textAlignVertical": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8C8C8C",
            "fontWeight": "bold",
          }
        }
      >
        Name
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f8f8f8",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Untitled"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#f8f8f8",
            "borderColor": "#f8f8f8",
            "borderWidth": 1,
            "color": "#8C8C8C",
            "flexDirection": "row",
            "fontWeight": "bold",
            "height": 53,
            "justifyContent": "space-between",
            "padding": 10,
            "paddingTop": 15,
            "textAlignVertical": "center",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "borderColor": "#f8f8f8",
          "borderWidth": 1,
          "fontWeight": "bold",
          "height": 53,
          "padding": 10,
          "paddingTop": 15,
          "textAlignVertical": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8C8C8C",
            "fontWeight": "bold",
          }
        }
      >
        Birthday
      </Text>
    </View>
    <MyDatePicker
      birthday="Not Set"
      setBirthday={[Function]}
    />
    <View
      style={
        Object {
          "backgroundColor": "#f8f8f8",
          "borderColor": "#f8f8f8",
          "borderWidth": 1,
          "color": "#8C8C8C",
          "flexDirection": "row",
          "fontWeight": "bold",
          "height": 53,
          "justifyContent": "space-between",
          "padding": 10,
          "paddingTop": 15,
          "textAlignVertical": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8C8C8C",
            "fontWeight": "bold",
          }
        }
      >
        Display Birthday
      </Text>
      <Switch
        ios_backgroundColor="gray"
        onValueChange={[Function]}
        style={
          Object {
            "alignItems": "center",
            "transform": Array [
              Object {
                "scaleX": 0.7,
              },
              Object {
                "scaleY": 0.6,
              },
            ],
          }
        }
        thumbColor="#f8f8f8"
        trackColor={
          Object {
            "false": "grey",
            "true": "#2FCC87",
          }
        }
        value={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "justifyContent": "flex-end",
        "width": "150%",
      }
    }
  >
    <ForwardRef
      disabled={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2FCC87",
          "borderRadius": 5,
          "color": "white",
          "padding": 15,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#92E1BF",
            "fontWeight": "bold",
          }
        }
      >
        Confirm
      </Text>
    </ForwardRef>
  </View>
</View>
`;
